checks/class-skip-links.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Check if skip links are present on block themes
+ *
+ * @package Theme Check
+ */
+
+/**
+ * Check if skip links are present on block themes
+ */
+class Skip_Links_Check implements themecheck {
+	/**
+	 * Error messages, warnings and info notices.
+	 *
+	 * @var array $error
+	 */
+	protected $error = array();
+
+	/**
+	 * Returns true if the theme is a block theme.
+	 *
+	 * @var array $is_block_theme
+	 */
+	protected $is_block_theme = false;
+
+	/**
+	 * The WP_Theme instance being checked.
+	 *
+	 * @var WP_Theme $wp_theme
+	 */
+	protected $wp_theme = false;
+
+	function set_context( $data ) {
+		if ( isset( $data['theme'] ) ) {
+			$this->wp_theme       = $data['theme'];
+			$this->is_block_theme = wp_is_block_theme();
+		}
+	}
+
+	/**
+	 * Check that return true for good/okay/acceptable, false for bad/not-okay/unacceptable.
+	 *
+	 * @param array $php_files File paths and content for PHP files.
+	 * @param array $css_files File paths and content for CSS files.
+	 * @param array $other_files Folder names, file paths and content for other files.
+	 */
+	public function check( $php_files, $css_files, $other_files ) {
+
+		$info                       = '';
+		$templates_without_main_tag = array();
+
+		$directory = 'templates'; // Path to the folder containing HTML files
+		$theme_dir = $this->wp_theme->get_stylesheet_directory();
+
+		// Get all HTML files in the directory
+		$files = glob( $theme_dir . '/' . $directory . '/*.html' );
```

Comment:
Accessing the file system from the `check` function doesn't seems to be necesary and it looks like it's not in line with the other implementation. The `check` function is already getting 3 arrays as parameters containing the list of file paths of the theme.